java/source/flight.rst

```diff
@@ -0,0 +1,1061 @@
+                            Result result = new Result("Delete completed".getBytes(StandardCharsets.UTF_8));
+                            listener.onNext(result);
+                    }
+                    listener.onCompleted();
+                }
+            }
+
+            @Override
+            public FlightInfo getFlightInfo(CallContext context, FlightDescriptor descriptor) {
+                if(!dataInMemory.containsKey(descriptor)){
+                    throw new IllegalStateException("Unknown descriptor.");
+                }
+                return new FlightInfo(
+                        dataInMemory.get(descriptor).getSchema(),
+                        descriptor,
+                        Collections.singletonList(new FlightEndpoint(new Ticket(descriptor.getPath().get(0).getBytes(StandardCharsets.UTF_8)), location)), // Configure a key to map back and forward your data using Ticket argument
+                        allocator.getAllocatedMemory(),
+                        dataInMemory.get(descriptor).getRows()
+                );
+            }
+
+            @Override
+            public void listFlights(CallContext context, Criteria criteria, StreamListener<FlightInfo> listener) {
+                dataInMemory.forEach((k, v) -> {
+                            FlightInfo flightInfo = getFlightInfo(null, k);
+                            listener.onNext(flightInfo);
+                        }
+                );
+                listener.onCompleted();
+            }
+        }).build();
+        try {
+            flightServer.start();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Client
+    try (RootAllocator allocator = new RootAllocator(Long.MAX_VALUE)){
+        // Populate data
+        FlightClient flightClient = FlightClient.builder(allocator, location).build();
+        Schema schema = new Schema(Arrays.asList( new Field("name", FieldType.nullable(new ArrowType.Utf8()), null)));
+        VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(schema, allocator);
+        VarCharVector varCharVector = (VarCharVector) vectorSchemaRoot.getVector("name");
+        varCharVector.allocateNew(3);
+        varCharVector.set(0, "Ronald".getBytes());
+        varCharVector.set(1, "David".getBytes());
+        varCharVector.set(2, "Francisco".getBytes());
+        varCharVector.setValueCount(3);
+        vectorSchemaRoot.setRowCount(3);
+        FlightClient.ClientStreamListener listener = flightClient.startPut(FlightDescriptor.path("profiles"), vectorSchemaRoot, new AsyncPutListener());
+        listener.putNext();
+        vectorSchemaRoot.allocateNew();
+        varCharVector.set(0, "Manuel".getBytes());
+        varCharVector.set(1, "Felipe".getBytes());
+        varCharVector.set(2, "JJ".getBytes());
+        varCharVector.setValueCount(3);
+        vectorSchemaRoot.setRowCount(3);
+        listener.putNext();
+        vectorSchemaRoot.clear();
```

Comment:
If `clear` is for memory management, it's clearer to use try-with-resources

Reply:
Added try-with-resources